internal/core/data/controller/messaging/subscriber.go

```diff
@@ -119,7 +120,10 @@ func validateEvent(messageTopic string, e dtos.Event) errors.EdgeX {
 	len := len(fields)
 	profileName := fields[len-3]
 	deviceName := fields[len-2]
-	sourceName := fields[len-1]
+	sourceName, err := url.QueryUnescape(fields[len-1])
```

Comment:
Don't we need similar changes in Core Metadata? Or why is this need if done in the middleware??

Reply:
This is used to unescape the message topic from the message bus because the device service publishes the event with the escaped resource name, see https://github.com/edgexfoundry/device-sdk-go/pull/1335/files#diff-ae127f9da6e44ed5a38a41c263f72c6e4d1b6da0a62a774df5861064d70e99a1R74

The middleware only unescapes the Url path variable.